### ISSUE TITLE
Is possible add new characters to the game?

### ISSUE THREAD

**Issue by kaioloureiro** [piratesephiroth/LaMulanaScript#1]
I'm trying to translate the game to portuguese, but when I try to add a new character, such as an "á", the in-game character appears with a blank space after it.

**Comment by piratesephiroth** [comment]
yes, you have to add then to the sheets as well.

Here's one example where I added characters from other languages
[pt.zip](https://github.com/piratesephiroth/LaMulanaScript/files/7999094/pt.zip)

**Comment by kaioloureiro** [comment]
Yeah, I know, but inside the game these new characters appears with a " " right after them, like shown below. It should be written "Ah, então você não quer?", instead of "Ah, entã o  você  nã o quer?". I need to solve it.

![lm](https://user-images.githubusercontent.com/22198551/152457672-b17fd0d6-f596-4805-ab54-8df0ae2f2d5f.png)

**Comment by piratesephiroth** [comment]
Ah, you have to set the width in grif.dat. I haven't made a tool for that but here's the one for the font I made

[grif.zip](https://github.com/piratesephiroth/LaMulanaScript/files/7999442/grif.zip)

**Comment by kaioloureiro** [comment]
Thank you! Worked smoothly. I think that your font has all the characters I need, but in case I need add something more in the future, can you teach me how to set the width on the grif.dat, please?

**Comment by piratesephiroth** [comment]
`grif.dat` is structured like I wrote in the readme.
It's a sequence of 16-bit values, big endian. The first value contains the glyph count, the rest contains some properties for each glyph:

1. The first byte is the horizontal offset (how many pixels to move to the right, if the image doesn't start right at the border of the 21x21 zone)
2. The second byte is the width of the character.

Here's some notes I kept here to help me edit the files:

```
To calculate char index from fontChars.txt:
index = count all selected chars up to the one you want - line breaks (each line break counts as 2 characters if using windows format)

To calculate the pixel position of the character in font00 from the index:
    x = (index % 48) * 21
    y = (index / 48) * 21

To calculate the index from pixel position in font00
    index = (x / 21) + ((y / 21) * 48)
```

**Comment by kaioloureiro** [comment]
I wasn't able to edit it. How can I open the grif.dat file and make these changes? The script doesn't do that.

**Comment by piratesephiroth** [comment]
yeah, as I said, I didn't make a tool for that.
You'll have to use a hex editor, like [HxD](https://mh-nexus.de/en/hxd/)
![image](https://user-images.githubusercontent.com/10672233/152477529-ae33912b-1e77-43e8-a8d4-d6a9fd84483f.png)

Specifically for editing grif.dat, you could set the byte group size to 2, to make it a bit easier to visualize.
![image](https://user-images.githubusercontent.com/10672233/152478504-b4066f5f-f745-49cf-aa41-ce3712edd8e7.png)

**Comment by Hexaae** [comment]
> Here's one example where I added characters from other languages [pt.zip](https://github.com/piratesephiroth/LaMulanaScript/files/7999094/pt.zip)
>...
> Ah, you have to set the width in grif.dat. **I haven't made a tool for that** but here's the one for the font I made
> 
> [grif.zip](https://github.com/piratesephiroth/LaMulanaScript/files/7999442/grif.zip)

🙏🏻Please do it... would be very helpful to solve many annoyances in some cases 😊

EDIT: however... in the end I was able to correctly add some Italian missing accented vowels for the intro/menu font, hex-editing the **grif.dat**. The original explanation wasn't that clear to me... so I suggest in addition to what's been said above:

- open grif.dat with a hex-editor grouped by 2 bytes (as already explained above)
- (as already explained above) **first two bytes in grif.dat are the total number of chars**, but the fontChars.txt from pt.zip above contains actually only 1265 chars (not $04ff=1279 as set in the first two bytes of grif.dat!), so if you want to add for example 20 chars you'll have to increase the number to $0505 (1265+20=1285). You can use free online chars counter web-apps like https://www.charactercountonline.com/ to verify the char number(s) (and positioning) **in the fontChars.txt**
- excluding the first 2 bytes in grif.dat, then every other 2 bytes groups are composed of: **$x-offset $x-length**, so f.e. $060b = 6 pixel offset from the left on the x axis, with length $0b (11), keeping in mind that each char-box is 21x21 pixels. Correct length is important or your char will have too much space after (game uses proportional fonts). Just add 1 pixel (so 10+1=11 in this example):

![image](https://github.com/piratesephiroth/LaMulanaScript/assets/8591462/51b4434c-08f2-4334-ada6-a263e32fe91b)
